### ISSUE TITLE
Ambiguous use of equal(to:) function

### ISSUE THREAD

**Issue by JesusMartinAlonso** [Brightify/Cuckoo#138]
I'm trying to stub a method that receives an objects that implements Equatable. The problem is that I get a

>  "Ambiguous use of equal(to:) function"

. The problem is that there are two functions that match with this argument. 

```
/// Returns an equality matcher.
equal<T: Equatable>(to value: T)

/// Returns an identity matcher.
equal<T: AnyObject>(to value: T)

```

**CODE:** 

```
       let mockedUser = User()
        mockedUser.id = "test"
        mockedUser.password = "testPassword"
    
        
        let mock = MockUserDataLayer()
      
        stub(mock) { stub in
            when(stub.createUser(user: equal(to:mockedUser))).thenDoNothing()
        }
```
How can I solve this?

**Comment by TadeasKriz** [comment]
Thank you for this report, definitely something that'll be removed before releasing 1.0.

For now you can use this workaround:
```
equal(to: $, equalWhen: ==)

equal(to: $, equalWhen: ===)
```

The first one is when you want to use `Equatable`, the second one when you want to check for the same instance. The second one should also be possible by `sameInstance(as: $)`, but there might be a missing method right now for non-nil parameters.

**Comment by JesusMartinAlonso** [comment]
Thanks for the clarification!

**Comment by TadeasKriz** [comment]
Fixed in `0.10.2`.
